### PR TITLE
Add editorconfig

### PR DESCRIPTION
**Pull request by jaythomas**
[editorconfig](http://editorconfig.org/) is a great way to set some file formatting conventions for a project and have your text editor automatically apply them. This saves the hassle of having to fix and re-review pull requests because of simple indentation issues or Windows-created files with CRLF line endings.

### CODE REVIEW

File: .editorconfig [hunk 1]
```diff
@@ -0,0 +1,14 @@
...
+
+# Match and apply these rules for all file
+# types you open in your code editor
+[*]
+# Unix-style newlines
+end_of_line = lf
+insert_final_newline = true
+indent_style = space
+indent_size = 4
+trim_trailing_whitespace = true
```
Comment (w0rp) [inline .editorconfig]: Add a trailing newline at the end of the file here... which this very same config seems to enable.

Comment (jaythomas) [reply .editorconfig]: hah yes I didn't have an editorconfig file when I created the file :grinning:

Comment (jaythomas) [reply .editorconfig]: Opening and re-saving the file, it automatically inserted it for me though. :+1:

**Comment by w0rp** [comment]
Cheers! :beers:

**Pull request merged**